File: SoftUni Lessons/Python Development/Python Basics April 2019/Lessons and Problems/07.1 - More Exercises Nested Conditional Statements/02. Bike Race .py
juniors_count = int(input())
seniors_count = int(input())
trace_type = input()

juniors_total = 0
seniors_total = 0

if trace_type == "trail":
    juniors_total = 5.50 * juniors_count
    seniors_total = 7 * seniors_count
elif trace_type == "cross-country":
    juniors_total = 8 * juniors_count
    seniors_total = 9.50 * seniors_count
elif trace_type == "downhill":
    juniors_total = 12.25 * juniors_count
    seniors_total = 13.75 * seniors_count
elif trace_type == "road":
    juniors_total = 20 * juniors_count
    seniors_total = 21.50 * seniors_count

total = juniors_total + seniors_total

if trace_type == "cross-country" and (juniors_count + seniors_count) >= 50:
    total -= total * 0.25

total -= total * 0.05

print(f"{total:.2f}")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/10. УПРАЖНЕНИЕ ПО-СЛОЖНИ ПРОВЕРКИ/Nested Conditional Statements - More Exercises/02. Bike Race.js
// Short link: https://bit.ly/2JhgeBP

function bikeRace(input) {
    let countJuniors = Number(input.shift());
    let countSeniors = Number(input.shift());
    let traceType = input.shift();

    let prices = {
        "trail": [5.5, 7],
        "cross-country": [8, 9.5],
        "downhill": [12.25, 13.75],
        "road": [20, 21.5]
    };

    let price = (prices[traceType][0] * countJuniors) + (prices[traceType][1] * countSeniors);

    if (traceType == "cross-country" && countJuniors + countSeniors >= 50) {
        price *= 0.75;
    }

    console.log(`${(price * 0.95).toFixed(2)}`);
}

bikeRace(['10', '20', 'trail']);  // Expected Output: 185.25
bikeRace(['20', '25', 'cross-country']);  // Expected Output: 377.63
bikeRace(['30', '25', 'cross-country']);  // Expected Output: 340.22
bikeRace(['10', '10', 'downhill']);  // Expected Output: 247.00
bikeRace(['3', '40', 'road']);  // Expected Output: 874.00
